passionscotch/app/models/user.rb

```diff
@@ -0,0 +1,42 @@
+require 'bcrypt'
+
+class User < ActiveRecord::Base
+  # Remember to create a migration!
+  has_many :reviews
+  has_many :wishlist
+
+
+
+  include BCrypt
+
+  validates :email, presence: true
+
+  def password
+    @password ||= Password.new(password_hash)
+  end
+
+  def password=(new_password)
+    @password = Password.create(new_password)
+    self.password_hash = @password
+  end
+
+  def create
+    @user = User.new(params[:user])
+    @user.password = params[:password]
+    @user.save!
+  end
+
+  def login
```

Comment:
the concept of logging in is a part of the web application not a part of your user model. There is a clear line here. The only concern a user model should have with authentication is confirming a password match.